### ISSUE TITLE
unsupported type: ptr

### ISSUE THREAD

**Issue by bgentry** [mitchellh/reflectwalk#8]
As described in https://github.com/mitchellh/copystructure/issues/2, I'm seeing this issue when using copystructure on Go 1.7:

```
panic: unsupported type: ptr [recovered]
	panic: unsupported type: ptr

goroutine 3050 [running]:
panic(0x421e40, 0xc420436580)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc4204386c0)
	/usr/local/go/src/testing/testing.go:579 +0x25d
panic(0x421e40, 0xc420436580)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/mitchellh/reflectwalk.walk(0x454020, 0xc420336988, 0x94, 0x4b7dc0, 0xc42049a000, 0x0, 0x0)
	/Users/bgentry/myproject/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:129 +0x251
github.com/mitchellh/reflectwalk.walkStruct(0x471ce0, 0xc420336980, 0x99, 0x4b7dc0, 0xc42049a000, 0x0, 0x8120d0)
	/Users/bgentry/myproject/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:264 +0x3ad
github.com/mitchellh/reflectwalk.walk(0x471ce0, 0xc420336980, 0x99, 0x4b7dc0, 0xc42049a000, 0x0, 0x0)
	/Users/bgentry/myproject/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:126 +0x311
github.com/mitchellh/reflectwalk.Walk(0x471ce0, 0xc420336980, 0x4b7dc0, 0xc42049a000, 0x471ce0, 0xc420336980)
	/Users/bgentry/myproject/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:70 +0x155
github.com/mitchellh/copystructure.Copy(0x471ce0, 0xc420336980, 0xc420336980, 0x471ce0, 0xc420336980, 0x1)
	/Users/bgentry/myproject/vendor/src/github.com/mitchellh/copystructure/copystructure.go:12 +0x64
```

If I add `reflect.Ptr` to the case statement on reflectwalk.go line 116:

```go
	case reflect.Bool, reflect.Chan, reflect.Func, reflect.Int, reflect.String, reflect.Invalid, reflect.Ptr:
```

then everything seems to work fine.

**Comment by bgentry** [comment]
actually, j/k, that doesn't fix it at all. The type of struct I'm trying to use this on comes from GRPC protobuf generation. It contains several levels of nested pointers, one of which is fulfilling an unexported interface (due to a protobuf `oneof` type).

I did try this other package and it seems to handle a deep copy on my struct with no issues: https://github.com/mohae/utilitybelt/blob/master/deepcopy/deepcopy.go

**Comment by mitchellh** [comment]
Fixed by #9
